python/ray/data/read_api.py

```diff
@@ -1659,13 +1661,44 @@ def read_tfrecords(
             By default, the number of output blocks is dynamically decided based on
             input data size and available resources. You shouldn't manually set this
             value in most cases.
-
+        fast_read_batch_size: An int representing the number of consecutive elements of
+            this dataset to combine in a single batch when fast_read is used.
+        fast_read_auto_infer_schema: Toggles the schema inference applied; applicable
+            only if fast_read is used and tf_schema argument is missing.
+            Defaults to True.
     Returns:
         A :class:`~ray.data.Dataset` that contains the example features.
 
     Raises:
         ValueError: If a file contains a message that isn't a ``tf.train.Example``.
     """
+    import platform
+
+    fast_read = False
+
+    try:
+        from tfx_bsl.cc.tfx_bsl_extension.coders import (  # noqa: F401
+            ExamplesToRecordBatchDecoder,
+        )
+
+        fast_read = True
+    except ModuleNotFoundError:
+        if platform.processor() == "arm":
+            logger.warning(
+                "The fast strategy of this function depends on tfx-bsl, which is "
+                "currently not supported on devices with Apple silicon "
+                "(e.g. M1) and requires an environment with x86 CPU architecture."
+            )
+        else:
+            logger.warning(
+                "To use TFRecordDatasource with large datasets, please install"
+                " tfx-bsl package with pip install tfx_bsl --no-dependencies`."
+            )
+        logger.info(
+            "Falling back to slower strategy for reading tf.records. This "
+            "reading strategy should be avoided when reading large datasets."
+        )
```

Comment:
Let's print out one warning message instead:

```
Please install tfx-bsl package with `pip install tfx_bsl --no-dependencies`. This can help speed up the reading of large TFRecord files.
```